src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+ * There is a call SetJewelColor that does nothing since 1.0f is passed.
+ * If a value of less than 1.0f were passed to SetJewelColor, then it would appear to drain the Jewel's color.
+ * This can be seen in preprelease screenshots.
+ */
+void DemoEffect_UpdateJewelAdult(DemoEffect* this, GlobalContext* globalCtx) {
+    this->jewel.timer++;
+    this->actor.shape.rot.y += 1024;
+    DemoEffect_PlayJewelSfx(this, globalCtx);
+    DemoEffect_SetJewelColor(this, 1.0f);
+}
+
+/**
+ * Update Function for the Jewel Actor that is run when Link is a child.
+ * This rotates the Jewel and updates a timer that is used to scroll Jewel textures.
+ * This also updates the Jewel's position based on different cutscenes.
+ */
+void DemoEffect_UpdateJewelChild(DemoEffect* this, GlobalContext* globalCtx) {
+    s32 hasCmdAction;
+    Actor* thisx = &this->actor;
+
+    this->jewel.timer++;
+
+    if (globalCtx->csCtx.state && globalCtx->csCtx.npcActions[this->csActionId]) {
+        switch (globalCtx->csCtx.npcActions[this->csActionId]->action) {
+            case 3:
+                if (gSaveContext.eventChkInf[4] & 2048) {
+                    gSaveContext.eventChkInf[4] |= 2048;
+                }
+                DemoEffect_MoveJewelActivateDoorOfTime(this, globalCtx);
+                if (!(globalCtx->gameplayFrames & 1)) {
+                    DemoEffect_JewelSparkle(this, globalCtx, 1);
+                }
+                break;
+            case 4:
+                if (this->jewel.isPositionInit) {
+                    DemoEffect_MoveToCsEndpoint(this, globalCtx, this->csActionId, 0);
+                    DemoEffect_MoveJewelSplit(&thisx->posRot, this);
+                    if (!(globalCtx->gameplayFrames & 1)) {
+                        DemoEffect_JewelSparkle(this, globalCtx, 1);
+                    }
+                } else {
+                    DemoEffect_InitPositionFromCsAction(this, globalCtx, this->csActionId);
+                    DemoEffect_MoveJewelSplit(&thisx->posRot, this);
+                    this->jewel.isPositionInit = 1;
+                }
+                break;
+            case 6:
+                Actor_Kill(thisx);
+                return;
+            default:
+                DemoEffect_MoveToCsEndpoint(this, globalCtx, this->csActionId, 0);
+                if (gSaveContext.entranceIndex == 0x0053) {
+                    DemoEffect_MoveJewelSplit(&thisx->posRot, this);
+                }
+                break;
+        }
+    }
+
+    if (gSaveContext.entranceIndex == 0x0053) {
+        if (!(gSaveContext.eventChkInf[4] & 2048)) {
+            hasCmdAction = globalCtx->csCtx.state && globalCtx->csCtx.npcActions[this->csActionId];
+            if (!hasCmdAction) {
+                this->effectFlags |= 1;
+                return;
+            }
+        }
+    }
+
+    thisx->shape.rot.y += 1024;
+    DemoEffect_PlayJewelSfx(this, globalCtx);
+    this->effectFlags &= 65534;
```

Comment:
```suggestion
    this->effectFlags &= ~1;
```

Reply:
Isn't this -2? 65535 would be -1

Reply:
Not `-2`, `~1`. Bitwise negation.

Reply:
Ah, my mistake. My phone is displaying it like a -1. You are right!